### DISCUSSION TITLE
MTD & FTD as a signle image

### DISCUSSION THREAD

**Discussion by 10andy0** [openthread/openthread#9839] [Q&A] [answered]
Hi there,
     Is there a way to compile a single binary with MTD & FTD support, and a way to chose mode in the code?
Thanks,
Andrii

**Comment by jwhui** [comment, marked as answer]
FTD build includes support to operate as an MTD - see [`otThreadSetLinkMode()`](https://openthread.io/reference/group/api-thread-general#otthreadsetlinkmode).
